### PR TITLE
ref: removed unused code for `atan2` intrinsic

### PR DESCRIPTION
**Pull request by parth121101**

### CODE REVIEW

File: src/libasr/pass/intrinsic_functions.h [hunk 1]
```diff
@@ -744,10 +655,70 @@ namespace Atan2 {
...
-        return BinaryIntrinsicFunction::instantiate_functions(al, loc, scope,
-            "atan2", arg_type, return_type, new_args, overload_id);
+
+
+        std::string c_func_name;
+        std::string new_name = "atan2";
+    switch (arg_type->type) {
+        case ASR::ttypeType::Complex : {
+            if (ASRUtils::extract_kind_from_ttype_t(arg_type) == 4) {
+                c_func_name = "_lfortran_c" + new_name;
```
Comment (Pranavchiku) [inline src/libasr/pass/intrinsic_functions.h:667]: I think this code is not properly indented, can you fix the indentation?

File: src/libasr/pass/intrinsic_functions.h [hunk 2]
```diff
@@ -744,10 +655,70 @@ namespace Atan2 {
...
+                c_func_name = "_lfortran_d" + new_name;
+            }
+        }
+    }
+    new_name = "_lcompilers_" + new_name + "_" + type_to_str_python(arg_type);
+
+    declare_basic_variables(new_name);
+    if (scope->get_symbol(new_name)) {
+        ASR::symbol_t *s = scope->get_symbol(new_name);
+        ASR::Function_t *f = ASR::down_cast<ASR::Function_t>(s);
```
Comment (Pranavchiku) [inline src/libasr/pass/intrinsic_functions.h:686]: Also do we not have a macro for these functions? I think we do.

File: src/libasr/pass/intrinsic_functions.h (hunk 1 above)
Comment (parth121101) [reply src/libasr/pass/intrinsic_functions.h:667]: Sure @Pranavchiku

**Review by Pranavchiku — COMMENTED** [review]
left comments above

**Review by certik — APPROVED** [review]

**Pull request merged**